Resources/doc/5-Usage-Event_Listener.md

```diff
@@ -1,75 +1,110 @@
-# Usage Sitemap Event Listeners
+# Sitemap Events Usage
 
-You can also register your sitemap event listeners by creating service classes implementing
-`Presta\SitemapBundle\Service\SitemapListenerInterface` and tagging these services with `presta.sitemap.listener`
-tag in your `Resources/config/services.xml`. This way the services will be lazy-loaded by Symfony's event dispatcher, only when the event is dispatched:
+You can also register event listeners (or subscribers) to populate your sitemap(s).
 
-```xml
-<parameters>
-    <parameter key="acme_demo.sitemap.listener.class">Acme\DemoBundle\EventListener\SitemapListener</parameter>
-</parameters>
+Imagine that your application is (or has) a blog, and that you want to add to your sitemap
+all blog posts that your administrator has created.
 
-<services>
-    <service id="my.sitemap.listener" class="%acme_demo.sitemap.listener.class%">
-        <tag name="presta.sitemap.listener" />
-        <argument type="service" id="router"/>
-    </service>
-</services>
-```
-
-or in yaml:
+**note :** we choose an `event subscriber` as example, but you can also do it with an `event listener`.
 
-```yaml
-parameters:
-    acme_demo.sitemap.listener.class: Acme\DemoBundle\EventListener\SitemapListener
 
-services:
-    my.sitemap.listener:
-        class: "%acme_demo.sitemap.listener.class%"
-        arguments: ["@router"]
-        tags: [{name: "presta.sitemap.listener"}]
-```
+## Service configuration
 
-Sitemap listener example `Acme/DemoBundle/EventListener/SitemapListener.php`:
+Implementation example `AppBundle/EventListener/SitemapBlogPostSubscriber.php`:
 
 ```php
 <?php
-namespace Acme\DemoBundle\EventListener;
 
-use Symfony\Component\Routing\RouterInterface;
-use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+namespace AppBundle\EventListener;
 
-use Presta\SitemapBundle\Service\SitemapListenerInterface;
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Routing\RouterInterface;
 use Presta\SitemapBundle\Event\SitemapPopulateEvent;
 use Presta\SitemapBundle\Sitemap\Url\UrlConcrete;
 
-class SitemapListener implements SitemapListenerInterface
+class SitemapBlogPostSubscriber implements EventSubscriberInterface
 {
+    /**
+     * @var RouterInterface
+     */
     private $router;
 
-    public function __construct(RouterInterface $router)
+    /**
+     * @var EntityManager
+     */
+    private $manager;
+
+    /**
+     * @param RouterInterface $router
+     * @param EntityManager   $manager
+     */
+    public function __construct(RouterInterface $router, EntityManager $manager)
```

Comment:
it is better use `UrlGeneratorInterface` and `EntityManagerInterface`. Also in all examples entity manager named like `$em`